[16.0][ADD] hr_planning_resources

antoniodavid opened the pull request.

rvalyi commented on hr_planning_resources/models/hr_leave.py:
```diff
@@ -0,0 +1,181 @@
+
+        def has_working_hours(start_dt, end_dt):
+            """
+            Returns `True` if there are any working days between `start_dt` and `end_dt`.
```
Should be imperative style https://stackoverflow.com/a/54868794
Same in previous docstring

antoniodavid replied:
Thanks for the tips!!!

victor-champonnois reviewed (commented):
Hi, thank you for the work!
I did a small functional exploration on the interface. 
- maybe a small doc on how to use the module would be nice
- I went on a project and clicked on "assign planning", there is a confirmation popup, and then nothing. What is this supposed to be doing ? 
![image](https://github.com/user-attachments/assets/6aacd717-d418-444e-bc48-93b90dfb5726)

FernandoRomera commented:
@antoniodavid 
Can you split this PR in two:
hr_planning_oca, only with menu app.
hr_planning_resources, all other.

antoniodavid commented:
> @antoniodavid Can you split this PR in two: hr_planning_oca, only with menu app. hr_planning_resources, all other.

Hi, could you explain the purpose of this feature?

antoniodavid committed:
[16.0][ADD] hr_planning_resources


[16.0][ADD] hr_planning_resources

[REF][16.0] hr_planning_resources: Refactoring app

[16.0][ADD] hr_planning_resources: Add readme

[16.0][ADD] hr_planning_resources: Add test-requirements

Revert "[16.0][ADD] hr_planning_resources: Add readme"

This reverts commit 4073ee9b9cf62c016e15b606d2d43f66266fecca.

[16.0][IMP] hr_planning_resources: Refactoring app

antoniodavid committed:
[DON'T MERGE] test-requirements.txt

antoniodavid commented:
> @antoniodavid Can you split this PR in two: hr_planning_oca, only with menu app. hr_planning_resources, all other.

Thanks for the advice!!!

rrebollo commented:
@antoniodavid @FernandoRomera why the split? I don't get it.

antoniodavid committed:
[16.0][IMP] hr_planning_resources: Pre-commit

christian-ramos-tecnativa commented:
I also don't think it's necessary to split it since the scope of the module is clear

antoniodavid committed:
[ADD][16.0] hr_planning_resources: Remove test-requirements

rrebollo commented:
> Depends on:
> 
> * [x]  [hr_planning_oca](https://github.com/OCA/resource/pull/4)

I think you can edit and mention this is not really needed anymore.

antoniodavid commented:
> > Depends on:
> > 
> > * [x]   [hr_planning_oca](https://github.com/OCA/resource/pull/4)
> 
> I think you can edit and mention this is not really needed anymore.

Fixed!!!!

jelenapoblet reviewed (approved):
On functional review, everything looks good, i was able to plan task, tickets and projects from the gantt view and viceversa.

antoniodavid commented:
ping @victor-champonnois @rvalyi

christian-ramos-tecnativa commented on hr_planning_resources/models/project_project.py:
```diff
@@ -0,0 +1,19 @@
+    @api.model_create_multi
+    def create(self, vals_list):
+        if self.env.context.get("default_user_id", False):
+            vals_list[0]["user_id"] = [(4, self.env.context["default_user_id"])]
```
```suggestion
            vals_list[0]["user_id"] = self.env.context["default_user_id"]
```
